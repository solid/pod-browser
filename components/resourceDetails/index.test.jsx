/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { waitFor } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";
import { mockSolidDatasetFrom } from "@inrupt/solid-client";
import { DatasetProvider } from "@inrupt/solid-ui-react";
import * as routerFns from "next/router";
import { renderWithTheme } from "../../__testUtils/withTheme";
import ResourceDetails, {
  TESTCAFE_ID_ACCORDION_PERMISSIONS,
  TESTCAFE_ID_ACCORDION_SHARING,
} from "./index";
import mockAccessControl from "../../__testUtils/mockAccessControl";
import { AccessControlProvider } from "../../src/contexts/accessControlContext";
import mockPermissionsContextProvider from "../../__testUtils/mockPermissionsContextProvider";
import useConsentBasedAccessForResource from "../../src/hooks/useConsentBasedAccessForResource";
import useAcp from "../../src/hooks/useAcp";
import useWac from "../../src/hooks/useWac";
import { TESTCAFE_ID_AGENT_ACCESS_TABLE } from "./resourceSharing/agentAccessTable";

const accessControl = mockAccessControl();
const dataset = mockSolidDatasetFrom("http://example.com/container/");

jest.mock("../../src/hooks/useConsentBasedAccessForResource");
const mockedUseConsentBasedAccessForResource = useConsentBasedAccessForResource;

jest.mock("../../src/hooks/useAcp");
jest.mock("../../src/hooks/useWac");
const mockUseAcp = useAcp;
const mockUseWac = useWac;

jest.mock("../../src/hooks/useConsentBasedAccessForResource");
const mockUseConsentBasedAccessForResource = useConsentBasedAccessForResource;

describe("Resource details", () => {
  beforeEach(() => {
    jest
      .spyOn(routerFns, "useRouter")
      .mockReturnValue({ query: { resourceIri: "" }, push: jest.fn() });
    mockedUseConsentBasedAccessForResource.mockReturnValue([]);
  });

  it("renders container details", async () => {
    mockUseAcp.mockReturnValue({ data: true });
    mockUseWac.mockReturnValue({ data: false });
    const { asFragment, getByText } = renderWithTheme(
      <DatasetProvider solidDataset={dataset}>
        <ResourceDetails />
      </DatasetProvider>
    );
    await waitFor(() => {
      expect(getByText("container")).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a decoded container name", async () => {
    mockUseAcp.mockReturnValue({ data: true });
    mockUseWac.mockReturnValue({ data: false });
    const datasetWithDecodedContainerName = mockSolidDatasetFrom(
      "http://example.com/Some%20container/"
    );

    const { asFragment, getByText } = renderWithTheme(
      <DatasetProvider solidDataset={datasetWithDecodedContainerName}>
        <ResourceDetails />
      </DatasetProvider>
    );
    await waitFor(() => {
      expect(getByText("Some container")).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Permissions component for WAC-supporting Solid servers", async () => {
    mockUseAcp.mockReturnValue({ data: false });
    mockUseWac.mockReturnValue({ data: true });
    const { asFragment, getByTestId, getByText } = renderWithTheme(
      <AccessControlProvider accessControl={accessControl}>
        <DatasetProvider solidDataset={dataset}>
          <ResourceDetails />
        </DatasetProvider>
      </AccessControlProvider>
    );
    await waitFor(() => {
      expect(getByTestId(TESTCAFE_ID_ACCORDION_PERMISSIONS)).toBeDefined();
      expect(getByText("Permissions")).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  }, 30000);

  it("renders Sharing component for ACP-supporting Solid servers", async () => {
    mockUseAcp.mockReturnValue({ data: true });
    mockUseWac.mockReturnValue({ data: false });
    const PermissionsContextProvider = mockPermissionsContextProvider();
    mockUseConsentBasedAccessForResource.mockReturnValue({ permissions: [] });
    const {
      asFragment,
      getByTestId,
      getByText,
      queryAllByTestId,
    } = renderWithTheme(
      <AccessControlProvider accessControl={accessControl}>
        <DatasetProvider solidDataset={dataset}>
          <PermissionsContextProvider>
            <ResourceDetails />
          </PermissionsContextProvider>
        </DatasetProvider>
      </AccessControlProvider>
    );
    await waitFor(() => {
      expect(getByTestId(TESTCAFE_ID_ACCORDION_SHARING)).toBeDefined();
      expect(getByText("Sharing")).toBeInTheDocument();
    });
    userEvent.click(getByText("Sharing"));
    await waitFor(() => {
      expect(queryAllByTestId(TESTCAFE_ID_AGENT_ACCESS_TABLE)).toHaveLength(2);
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
