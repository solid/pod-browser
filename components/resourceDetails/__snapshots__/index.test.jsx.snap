// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource details renders Permissions component for WAC-supporting Solid servers 1`] = `
<DocumentFragment>
  <section
    class="PodBrowser-headerSection"
  >
    <h3
      class="PodBrowser-content-h3"
      data-testid="resource-title"
      title="http://example.com/container/"
    >
      container
    </h3>
  </section>
  <hr
    class="PodBrowser-root"
  />
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
      data-testid="accordion-resource-actions-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content PodBrowser-expanded"
      >
        Actions
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <ul
                class="PodBrowser-action-menu"
              >
                <li
                  class="PodBrowser-action-menu__item"
                />
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger"
                    data-testid="resource-address-copy-tooltip"
                    type="button"
                  >
                    COPY LINK
                  </button>
                </li>
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger PodBrowser-action-menu__trigger--danger"
                    data-testid="delete-resource-button"
                    type="button"
                  >
                    Delete
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="PodBrowser-root PodBrowser-root"
      data-testid="accordion-resource-details-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content"
      >
        Details
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-hidden"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <section
                class="PodBrowser-centeredSection"
              >
                <ul
                  class="PodBrowser-root PodBrowser-padding"
                >
                  <li
                    class="PodBrowser-root PodBrowser-gutters"
                  >
                    <p
                      class="PodBrowser-root PodBrowser-detailText PodBrowser-body1"
                    >
                      Thing Type:
                    </p>
                    <p
                      class="PodBrowser-root PodBrowser-typeValue PodBrowser-detailText PodBrowser-body1"
                    >
                      text/turtle
                    </p>
                  </li>
                </ul>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="PodBrowser-root PodBrowser-root"
      data-testid="accordion-resource-permissions"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content"
      >
        Permissions
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-hidden"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <ul
                class="PodBrowser-action-menu"
              >
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-button PodBrowser-button--small"
                    data-testid="permissions-add-user-with-webid-button"
                  >
                    <span
                      class="PodBrowser-button__text"
                    >
                      Add with WebId
                    </span>
                    <span
                      class="PodBrowser-icon-caret-right PodBrowser-button__icon PodBrowser-button__icon--after"
                    />
                  </button>
                </li>
              </ul>
              <div
                class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
              >
                <div
                  aria-disabled="false"
                  aria-expanded="true"
                  class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="PodBrowser-content PodBrowser-expanded"
                  >
                    Individual permissions
                  </div>
                  <div
                    aria-disabled="false"
                    aria-hidden="true"
                    class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
                  >
                    <span
                      class="PodBrowser-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="PodBrowser-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </span>
                    <span
                      class="PodBrowser-root"
                    />
                  </div>
                </div>
                <div
                  class="PodBrowser-container PodBrowser-entered"
                  style="min-height: 0px;"
                >
                  <div
                    class="PodBrowser-wrapper"
                  >
                    <div
                      class="PodBrowser-wrapperInner"
                    >
                      <div
                        role="region"
                      >
                        <div
                          class="PodBrowser-root PodBrowser-details"
                        >
                          <div>
                            No access granted to individuals yet
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Resource details renders Sharing component for ACP-supporting Solid servers 1`] = `
<DocumentFragment>
  <section
    class="PodBrowser-headerSection"
  >
    <h3
      class="PodBrowser-content-h3"
      data-testid="resource-title"
      title="http://example.com/container/"
    >
      container
    </h3>
  </section>
  <hr
    class="PodBrowser-root"
  />
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
      data-testid="accordion-resource-actions-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content PodBrowser-expanded"
      >
        Actions
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <ul
                class="PodBrowser-action-menu"
              >
                <li
                  class="PodBrowser-action-menu__item"
                />
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger"
                    data-testid="resource-address-copy-tooltip"
                    type="button"
                  >
                    COPY LINK
                  </button>
                </li>
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger PodBrowser-action-menu__trigger--danger"
                    data-testid="delete-resource-button"
                    type="button"
                  >
                    Delete
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="PodBrowser-root PodBrowser-root"
      data-testid="accordion-resource-details-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content"
      >
        Details
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-hidden"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <section
                class="PodBrowser-centeredSection"
              >
                <ul
                  class="PodBrowser-root PodBrowser-padding"
                >
                  <li
                    class="PodBrowser-root PodBrowser-gutters"
                  >
                    <p
                      class="PodBrowser-root PodBrowser-detailText PodBrowser-body1"
                    >
                      Thing Type:
                    </p>
                    <p
                      class="PodBrowser-root PodBrowser-typeValue PodBrowser-detailText PodBrowser-body1"
                    >
                      text/turtle
                    </p>
                  </li>
                </ul>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
      data-testid="accordion-resource-sharing"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content PodBrowser-expanded"
      >
        Sharing
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px; height: auto; transition-duration: 0ms;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <div
                class="PodBrowser-root PodBrowser-root PodBrowser-accordion PodBrowser-expanded PodBrowser-rounded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
                data-testid="agent-access-table"
              >
                <div
                  class="PodBrowser-headerContainer"
                >
                  <i
                    class="PodBrowser-icon-editor PodBrowser-icon PodBrowser-iconEditor"
                  />
                  <div
                    class="PodBrowser-textContainer"
                  >
                    <div
                      class="PodBrowser-titleAndButtonContainer"
                    >
                      <p
                        class="PodBrowser-title"
                      >
                        Editors
                      </p>
                      <button
                        class="PodBrowser-button PodBrowser-button--text"
                        data-testid="add-agent-button"
                      >
                        <span
                          class="PodBrowser-icon-edit PodBrowser-button__icon PodBrowser-button__icon--before"
                        />
                        <span
                          class="PodBrowser-button__text"
                        >
                          Edit Editors
                        </span>
                      </button>
                      <button
                        class="PodBrowser-button PodBrowser-button--icon-only PodBrowser-action-button"
                      >
                        <span
                          class="PodBrowser-button__text"
                        >
                          <span
                            aria-label="Show menu for policy"
                            class="PodBrowser-icon-more"
                          />
                        </span>
                      </button>
                    </div>
                    <span
                      class="PodBrowser-description"
                    >
                      <p>
                        <b>
                          Can 
                        </b>
                        view, edit and delete this resource
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  class="PodBrowser-container PodBrowser-entered"
                  style="min-height: 0px;"
                >
                  <div
                    class="PodBrowser-wrapper"
                  >
                    <div
                      class="PodBrowser-wrapperInner"
                    >
                      <div
                        role="region"
                      >
                        <div
                          class="PodBrowser-permissionsContainer"
                        >
                          <span
                            class="PodBrowser-emptyStateTextContainer"
                          >
                            <p>
                              No editors
                            </p>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="PodBrowser-root PodBrowser-root PodBrowser-accordion PodBrowser-expanded PodBrowser-rounded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
                data-testid="agent-access-table"
              >
                <div
                  class="PodBrowser-headerContainer"
                >
                  <i
                    class="PodBrowser-icon-view PodBrowser-icon PodBrowser-iconViewer"
                  />
                  <div
                    class="PodBrowser-textContainer"
                  >
                    <div
                      class="PodBrowser-titleAndButtonContainer"
                    >
                      <p
                        class="PodBrowser-title"
                      >
                        Viewers
                      </p>
                      <button
                        class="PodBrowser-button PodBrowser-button--text"
                        data-testid="add-agent-button"
                      >
                        <span
                          class="PodBrowser-icon-edit PodBrowser-button__icon PodBrowser-button__icon--before"
                        />
                        <span
                          class="PodBrowser-button__text"
                        >
                          Edit Viewers
                        </span>
                      </button>
                      <button
                        class="PodBrowser-button PodBrowser-button--icon-only PodBrowser-action-button"
                      >
                        <span
                          class="PodBrowser-button__text"
                        >
                          <span
                            aria-label="Show menu for policy"
                            class="PodBrowser-icon-more"
                          />
                        </span>
                      </button>
                    </div>
                    <span
                      class="PodBrowser-description"
                    >
                      <p>
                        <b>
                          Can 
                        </b>
                        view but
                        <b>
                           cannot 
                        </b>
                        edit or delete this resource
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  class="PodBrowser-container PodBrowser-entered"
                  style="min-height: 0px;"
                >
                  <div
                    class="PodBrowser-wrapper"
                  >
                    <div
                      class="PodBrowser-wrapperInner"
                    >
                      <div
                        role="region"
                      >
                        <div
                          class="PodBrowser-permissionsContainer"
                        >
                          <span
                            class="PodBrowser-emptyStateTextContainer"
                          >
                            <p>
                              No viewers
                            </p>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="PodBrowser-sharingButtonContainer"
              >
                <button
                  class="PodBrowser-button PodBrowser-button--text PodBrowser-sharingButton"
                  data-testid="advanced-sharing-button"
                >
                  <span
                    class="PodBrowser-button__text"
                  >
                    <i
                      class="PodBrowser-icon-settings PodBrowser-icon"
                    />
                    Advanced Sharing
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Resource details renders a decoded container name 1`] = `
<DocumentFragment>
  <section
    class="PodBrowser-headerSection"
  >
    <h3
      class="PodBrowser-content-h3"
      data-testid="resource-title"
      title="http://example.com/Some%20container/"
    >
      Some container
    </h3>
  </section>
  <hr
    class="PodBrowser-root"
  />
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
      data-testid="accordion-resource-actions-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content PodBrowser-expanded"
      >
        Actions
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <ul
                class="PodBrowser-action-menu"
              >
                <li
                  class="PodBrowser-action-menu__item"
                />
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger"
                    data-testid="resource-address-copy-tooltip"
                    type="button"
                  >
                    COPY LINK
                  </button>
                </li>
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger PodBrowser-action-menu__trigger--danger"
                    data-testid="delete-resource-button"
                    type="button"
                  >
                    Delete
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="PodBrowser-root PodBrowser-root"
      data-testid="accordion-resource-details-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content"
      >
        Details
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-hidden"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <section
                class="PodBrowser-centeredSection"
              >
                <ul
                  class="PodBrowser-root PodBrowser-padding"
                >
                  <li
                    class="PodBrowser-root PodBrowser-gutters"
                  >
                    <p
                      class="PodBrowser-root PodBrowser-detailText PodBrowser-body1"
                    >
                      Thing Type:
                    </p>
                    <p
                      class="PodBrowser-root PodBrowser-typeValue PodBrowser-detailText PodBrowser-body1"
                    >
                      text/turtle
                    </p>
                  </li>
                </ul>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Resource details renders container details 1`] = `
<DocumentFragment>
  <section
    class="PodBrowser-headerSection"
  >
    <h3
      class="PodBrowser-content-h3"
      data-testid="resource-title"
      title="http://example.com/container/"
    >
      container
    </h3>
  </section>
  <hr
    class="PodBrowser-root"
  />
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-expanded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="PodBrowser-root PodBrowser-root PodBrowser-expanded"
      data-testid="accordion-resource-actions-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content PodBrowser-expanded"
      >
        Actions
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-expanded PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <ul
                class="PodBrowser-action-menu"
              >
                <li
                  class="PodBrowser-action-menu__item"
                />
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger"
                    data-testid="resource-address-copy-tooltip"
                    type="button"
                  >
                    COPY LINK
                  </button>
                </li>
                <li
                  class="PodBrowser-action-menu__item"
                >
                  <button
                    class="PodBrowser-action-menu__trigger PodBrowser-action-menu__trigger--danger"
                    data-testid="delete-resource-button"
                    type="button"
                  >
                    Delete
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="PodBrowser-root PodBrowser-root"
      data-testid="accordion-resource-details-trigger"
      role="button"
      tabindex="0"
    >
      <div
        class="PodBrowser-content"
      >
        Details
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="PodBrowser-root PodBrowser-root PodBrowser-expandIcon PodBrowser-edgeEnd"
      >
        <span
          class="PodBrowser-label"
        >
          <svg
            aria-hidden="true"
            class="PodBrowser-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="PodBrowser-root"
        />
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-hidden"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-root PodBrowser-accordionDetails"
            >
              <section
                class="PodBrowser-centeredSection"
              >
                <ul
                  class="PodBrowser-root PodBrowser-padding"
                >
                  <li
                    class="PodBrowser-root PodBrowser-gutters"
                  >
                    <p
                      class="PodBrowser-root PodBrowser-detailText PodBrowser-body1"
                    >
                      Thing Type:
                    </p>
                    <p
                      class="PodBrowser-root PodBrowser-typeValue PodBrowser-detailText PodBrowser-body1"
                    >
                      text/turtle
                    </p>
                  </li>
                </ul>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
